File: No 06 - DT-Training/Chapter02/EveryoneLovesEF/Intro/Model/Player.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Intro.Model
{
    public class Player
    {
        public int PlayerId { get; set; }
        [StringLength(12)]
        [Required]
        [Column("nick_name")]
        public string Nickname { get; set; }
        [Required]
        [Column("current_level")]
        public short Level { get; set; }
    }
}

File: No 23 - A Simple React App/src/gamer/src/models/player.ts
/*
    Model nesnemiz.
    Veritabanı ile etkileşeme geçecek diğer tiplerde IPlayer sözleşmesine uyan bu modeli kullanabileceğiz
*/
import { IPlayer } from "./../types/player"
import { model, Schema } from "mongoose"

const playerSchema: Schema = new Schema(
  {
    nickname: {
      type: String,
      required: true,
    },

    country: {
      type: String,
      required: true,
    },

    level: {
      type: Number,
      required: true,
    },
  },
  { timestamps: true }
)

export default model<IPlayer>("Player", playerSchema)
